### DISCUSSION TITLE
useStepsForm before submission event using ant-design UI

### DISCUSSION THREAD

**Discussion by heriparid** [refinedev/refine#2429] [Q&A] [answered]
Hi Team,

Currently I'm using step wizard using `useStepForm` using ant design UI
I have encountered some problems on how to get values before submitting to backend. I have a use case to implement some logic i.e: value-encryption. So that when the data is submitted to backend, the data is already encrypted.

First Try
I have tried to make use of "onFinish" event in my <Form> component, but the onFinish function is not triggered when I click save/submit event. It's simply post the data directly to backend

Second Try
Using Form Instance as returned from useStepForm, and then by using formInstance, I invoke `formProps.form?.submit();` then my onFinish function event is triggered, but unfortunately, only the last array of form values only populated

**Comment by omeraplak** [comment, marked as answer]
Hey @heriparid 👋,
Thank you for contacting us! You can override submit function of useStepsForm hook ⚡️

```tsx
const stepsForm = useStepsForm({
    submit: (values) => {
        console.log(values);
    },
});
```

**Reply by heriparid** [reply]
Hi @omeraplak ,
Thanks for giving me a feedback.
I just add submit function implementation in the useStepForm. But unfortunately, it's not triggred when I click save button.

My save button implementation is like this:
```
const { stepsProps, formProps, current, gotoStep, saveButtonProps } =
    useStepsForm<ISendMoneyInquiryResponse>({
      submit: (values) => {
        console.log(JSON.stringify(values));
      },

...
<SaveButton {...saveButtonProps} />
```
Am I missing something?

**Reply by omeraplak** [reply]
Can you try our useStepsForm example?
https://refine.dev/docs/examples/form/antd/useStepsForm/

**Reply by heriparid** [reply]
Hi @omeraplak ,
I just saw and tried to edit in the built in IDE on the link given by you.
The `submit` function is successfully triggered as expected
Here's what I see
![image](https://user-images.githubusercontent.com/3928060/188278298-3c0a5ea6-0eab-48b6-a7f5-4eb2876d1d9e.png)

It's weird, I don't have a clue to spot what is going wrong on my own code :(
The save button is still not triggering submit function

**Reply by omeraplak** [reply]
Interesting... Are you sure all your refine dependencies are the latest version?

**Reply by heriparid** [reply]
Hi @omeraplak ,
Having updated the dependencies, pyuh,, It's working now :)
![image](https://user-images.githubusercontent.com/3928060/188279493-38426f26-84d6-4b09-b858-bd8d558a3fd6.png)
Thanks for guiding me.

Cheers

**Reply by omeraplak** [reply]
Cheers 🍻

**Reply by omeraplak** [reply]
Don't forget to support us by giving stars!  thanks 🤩
